src/interpreter/index.ts

```diff
@@ -94,6 +96,45 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         this.events.once("err", errorHandler);
     }
 
+    /**
+     * Builds out all the sub-environments for the given components. Components are saved into the calling interpreter
+     * instance. This function will mutate the state of the calling interpreter.
+     * @param componentMap Map of all components to be assigned to this interpreter
+     * @param manifest
+     * @param formatPathFn
+     * @param parseFn
+     */
+    public static async withSubEnvsFromComponents(
+        componentMap: Map<string, ComponentDefinition>,
+        formatPathFn: (comp: ComponentDefinition) => void,
+        parseFn: (filenames: string[]) => Promise<Stmt.Statement[]>,
+        options: ExecutionOptions = defaultExecutionOptions
+    ) {
+        let interpreter = new Interpreter(options);
+        interpreter.onError(logError);
+
+        let componentDefinitions = (interpreter.environment.nodeDefMap = componentMap);
+        if (componentDefinitions.size > 0) {
```

Comment:
This seems redundant as Promise.all() and Array.from() should handle empty arrays.